llvm/test/Transforms/Attributor/nofpclass-frexp.ll

```diff
@@ -27,8 +27,8 @@ define { float, i32 } @ret_frexp_f32_nonan(float nofpclass(nan) %arg0) {
 
 define float @ret_frexp_f32_0_nonan(float nofpclass(nan) %arg0) {
 ; CHECK-LABEL: define nofpclass(nan sub) float @ret_frexp_f32_0_nonan
-; CHECK-SAME: (float nofpclass(nan sub) [[ARG0:%.*]]) #[[ATTR1]] {
-; CHECK-NEXT:    [[CALL:%.*]] = call { float, i32 } @llvm.frexp.f32.i32(float [[ARG0]]) #[[ATTR7]]
+; CHECK-SAME: (float nofpclass(nan) [[ARG0:%.*]]) #[[ATTR1]] {
```

Comment:
I think we still need the special case for frexp, this should still imply nofpclass(sub)

Reply:
What is the rule then?
This would work already if frexp had the proper argument attribute.

Reply:
You cannot apply the attribute to the return value because it's a struct, not an FP/FP vector. You have to specially recognize the extractvalue of the first element of the struct

Reply:
I don't get it. Extractvalue does not imply anything.
What you said first, or what I understood, is that frexp implies that the argument is `nofpclass(sub)`. Is that true? If so, I can implement that.

Reply:
The floating point component of the return value of frexp cannot be a denormal. The input could be anything. Because it returns a struct, anything you want to say about the frexp result has to be applied to the extractvalue user of frexp, not the frexp directly

Reply:
The test changed such that the argument of the function, the argument to frexp, is not assumed non-sub. What is the problem here?

Reply:
> What is the problem here?

That I'm reading the argument attribute as the return